docs/modules/ROOT/pages/explanations/decisions/deletion-protection.adoc

```diff
@@ -0,0 +1,129 @@
+= Deletion Protection
+
+== Problem
+
+To ensure that no PVCs and buckets are deleted without explicit intention, we need various systems that create deletion protection at different levels.
+There are multiple layers of deletion protection to consider:
+
+* Accidental deletion by an end-user -> for example deleting a claim without intending to
+* Accidental deletion on the backend -> for example a bug in the composition-function unintentionally removes something
+
+Currently, there's a deletion protection for PostgreSQL in place with a separate controller.
+However, this controller only protects against accidental deletion of the claim, nothing else.
+It also lacks support for other services and contains a few bugs which prevent it from cleaning up the resources after the grace period.
+
+=== Goals
+
+* Easy to use end-user deletion protection
+* Easy to extend and integrate backend deletion protection
+
+=== Non-Goals
+
+* N/A
+
+== Proposals
+The proposals are split between the end-user and backend deletion protection.
+
+== End-user Proposal 1: Protect claim deletion via webhooks
```

Comment:
The deletion protection was a topic way before we had webhooks. But yeah in retrospection this would have been a much better solution. Finalizers are more intended to ensure some ordering/cleanup after the object has already been deleted.